### PR TITLE
Alternative usage, more similar to Google Mock

### LINKED ISSUE #9: Suggested alternative usage, more similar to Google Mock
Hello,
I was shown C-Mock and find it a very useful extension to Google Mock.
When using it, I was a little annoyed by the need to define a separate mock class for each mocked function.
I believe, based on reading the code and my experience in messing with Google Mock under the hood, that better usage is possible, and I can implement it.

Here's my suggestion:

In the header file, declare a mock class, similarly to Google Mock, where each mocked function is a method:
```
class FooFunctionMock : FunctionMocker<FooFunctionMock>
{
public:
     MOCK_FUNCTION2(foo, int, (int, int));
     MOCK_FUNCTION1(bar, int, (int));
};
```
The `FunctionMocker` class will allow access to the `FooFunctionMock` instance (similarly to the `instance` pointer).
`MOCK_FUNCTIONn` will be similar to `MOCK_METHODn`, but will declare a `real_fname` member and `call_fname` method.

In the mock source file, use `IMPLEMENT_FUNCTION_MOCK` similarly to today:
```
IMPLEMENT_FUNCTION_MOCK2(FooFunctionMock, foo, int(int, int)); 
IMPLEMENT_FUNCTION_MOCK1(FooFunctionMock, bar, int(int));
```
Unlike the current macro, it won't implement a constructor and destructor, just `real_fname` and `call_fname`. So there's no problem in using it multiple times with the same class name.
I guess we'll need to rename it, to avoid conflict with the existing macro.

Expectations will be set using `EXPECT_CALL`, just like any Google Mock class.

### PR DESCRIPTION
**Pull request by ugoren**
Allow creating a Google Mock class with multiple methods, and redirecting several function calls to the same class.
Resolves #9

### CODE REVIEW

File: test/foo_mock_class.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#include "foo.h"
+
+class FooMocker : public CMockMocker<FooMocker> 
+{
+public:
+    MOCK_METHOD2(add, int(int, int));
+    MOCK_METHOD2(sub, int(int, int));
+};
+
+class CountMocker : public CMockMocker<CountMocker>
```
Comment (hjagodzinski) [inline test/foo_mock_class.h]: `CountMocker` is declared but not used therefore should be removed.

**Comment by hjagodzinski** [comment]
What's lacking is a documentation update. I will handle this myself as I want to rewrite README to be more easy to follow.

**Comment by hjagodzinski** [comment]
I've configured travis integration. Could you merge your branch with the current master?

**Comment by hjagodzinski** [comment]
Thanks!

**Comment by ugoren** [comment]
Thanks for merging.
We already started using the new method in my company, and now we'll be
able to switch to the master branch.

BTW, I have a pull request to Google Mock as well (
https://github.com/google/googletest/pull/1532) which may affect C-Mock.
It add support for MOCK_METHOD, without an argument count.
If it's ever merged (getting stuff merged to googletest is not simple...)
we'll be able to add similar functionality to C-Mock macros.

**Comment by hjagodzinski** [comment]
It' really cool any one actually find this project helpful 😎

Fingers crossed your contribution will finally end up in the master branch 🤞

I've created a new release v0.2.0 with your contribution:
* README includes information on *CMockMocker* usage
* the old macros are marked as deprecated.

**Comment by devesh25427** [comment]
How can I call real function if I am using new style CMOCK_MOCK_FUNCTIONn function?

**Pull request merged**